### ISSUE TITLE
False-positive in CmdCopyQueryPoolResults

### ISSUE THREAD

**Issue by zeux** [KhronosGroup/Vulkan-LoaderAndValidationLayers#722]
In my application, I have 1 QueryPool with N queries and K command buffers.

I use CmdWriteTimestamp on K-1 command buffers (except for the last one) to fill first N-1 queries - each query is only written in one command buffer, of course. Then I submit all command buffers to the graphics queue.

After this I add one more CmdWriteTimestamp (for the last query), followed by CmdCopyQueryPoolResults and CmdResetQueryPool to the last command buffer, and submit it to the same queue.

This submission results in N-1 warnings - one for every query *not* written in the last command buffer - that the query is invalid. This leads me to believe that CmdCopyQueryPoolResults only considers queries written in the same command buffer as valid.

As far as I understand this is not part of the spec - I should be able to use any number of command buffers to write the data as long as they are all submitted before CopyQueryPoolResults is.

GetQueryPoolResults does not have this problem - I think it's because it goes through all in-flight command buffers, not through just one of them, to find the query state.

**Comment by zeux** [comment]
The specific error message looks like this:

```
ERROR: [DS] Code 14 : Requesting a copy from query to buffer with invalid query: queryPool 0x67, index 114
```

**Comment by zeux** [comment]
The original post is not quite exact. The more exact description is that the command buffers that the timestamps are being ignored in are secondary. That is:

I have K-2 secondary command buffers with timestamp writes for N-1 queries;
I have 1 primary command buffer with CmdExecuteCommands(K-2)
I have 1 primary command buffer with 1 CmdWriteTimestamp, CmdCopyQueryPoolResults and CmdResetQueryPool.

So it's possible that instead of not considering other command buffers CopyQueryPoolResults just does not work with queries submitted by secondary command buffers? I'm not familiar with the codebase for the validation layers to tell.

**Comment by tobine** [comment]
Fix was cherry-picked and pushed to master
